src/diff.ts

```diff
@@ -0,0 +1,72 @@
+import parseDiff, { AddChange } from 'parse-diff'
+import { LcovFile } from 'lcov-parse'
+import { GitHub } from '@actions/github/lib/utils'
+import { ContextInfo } from './types'
+
+interface Props {
+  lcovFiles: LcovFile[]
+  diffString: string
+  octokit: InstanceType<typeof GitHub>
+  context: ContextInfo
+}
+
+export const analyzeDiffCoverage = async ({
```

Comment:
It would be really nice to also include some tests for this function. You could extract the diff analysis code and separate it from the github integration. This would make it much easier to test, without the need to use mocks.